### DISCUSSION TITLE
Ability to redact multiple messages

### DISCUSSION THREAD

**Discussion by aurabindo** [element-hq/element-meta#921] [Ideas]
When there are lots of messages, it is inconvenient to delete them one by one.

**Comment by lampholder** [comment]
This raises an interesting question - how easy do we want to make redacting?

I guess what we should consider is a mechanism for 'selecting' multiple messages; then you can quote them, redact them, forward them etc.

**Comment by pafcu** [comment]
Related: https://github.com/vector-im/riot-web/issues/5536

**Comment by lampholder** [comment]
Related:
- https://github.com/vector-im/riot-web/issues/2158 
- https://github.com/vector-im/riot-web/issues/3689
- https://github.com/vector-im/riot-web/issues/5885
- https://github.com/vector-im/riot-web/issues/5898

**Comment by PC-Admin** [comment]
Why would anyone want it to be that hard?

Riot is currently quite annoying to use because of this. Want to delete 5 messages? You are going to have to click 15 times. 15. It's a cruel joke.

You should be able to select multiple messages (or all in a room) then redact them with 3 clicks. 

Can we please make this happen? The Riot UI/UX is currently bad and i would like to see Riot become something worth telling people about.

**Comment by Croydon** [comment]
Since we are just getting in a lot of spam from Freenode, I would really love to see this happening.

**Comment by rubo77** [comment]
Limit it to IRC bridge users and only to users who already left

**Comment by jywarren** [comment]
Much agreed! We're getting hit hard in https://riot.im/app/#/room/#publiclab-testing:matrix.org (warning ugly content)

on the order of hundreds of messages from 10s of spambot users. 

Any way for us to help out?

**Comment by nunoperalta** [comment]
I was looking into this suggestion. I hope it's still being considered.

**Comment by rubo77** [comment]
put a +1 on the first post to vote for this feature

**Comment by MurzNN** [comment]
Good example of UI for multiple select messages (for delete/reply/forward) we can lookup in https://vk.com/messenger app:
![image](https://user-images.githubusercontent.com/336662/61691716-d54cde00-ad34-11e9-9bf9-b5b83da1f255.png)

**Comment by n1trux** [comment]
Why is this major papercut p3 and open since 2017?

**Comment by ara4n** [comment]
a) it has not been a higher priority because matrix's rate limiting means it's relatively unusual to end up with large amounts of spam in a room (as spammers can't spam quickly, and you can redact as fast as they spam). it's also worth noting that other messengers (e.g. WhatsApp) don't have this feature, and Telegram only sprouted it relatively recently (in the last year).  it's become more of a priority over the last few months, and the priority label hasn't been adjusted to reflect that.
b) this *is* a new feature.
c) it's being worked on right now at https://github.com/matrix-org/matrix-doc/pull/2244.

**Comment by n1trux** [comment]
Thank you for making this p1. Excuse my snark.

**Comment by grinapo** [comment]
> a) it has not been a higher priority because matrix's rate limiting means it's relatively unusual to end up with large amounts of spam in a room (as spammers can't spam quickly, and you can redact as fast as they spam).

…unless you sleep and in the morning there are 378 messages waiting with female genitalia. :-) That'd be 1890 clicks.

> c) it's being worked on right now at [matrix-org/matrix-doc#2244](https://github.com/matrix-org/matrix-doc/pull/2244).

Yep, let's watch not to hit rate limiting with redactions, until gets 2244 implemented.

**Comment by n1trux** [comment]
vector-im/element-web#7660 _[Feature Request] Add "clear chat history" option or multiple message selection_

seems to be related.

**Comment by damascene** [comment]
Any update on this issue. Still hard to delete multiple messages one by one. Please add select option.

**Comment by rubo77** [comment]
As a workaround use the spam-fighting bot:
https://github.com/matrix-org/mjolnir

**Comment by gravax** [comment]
Still no update... Deleting 86 messages from 86 fake spamming accounts that all arrived within 3 minutes of each other this morning... That's 5 clicks for each (remove recent messages, ban)...

**Comment by alittlebitofit** [comment]
@gravax damn thats so annoying and such a waste of time

developers please make this a highest priority already, for all platforms
Android, Web, etc

1-multiselect
2-multicopy
3-multidelete
4-multishare
5-other things that can be done with multi feature that i dont recall as of now

thank you

**Comment by dimyself** [comment]
why do all these big sites/apps have all this useless crap features that no one hardly cares about...but main vital features aren't available??? 

I have a conversation with someone and i need to delete like 200 latest msgs, but i cant! 

This shouldn't even be a hard feauture to implement! why since 2017 this isn't available??? after 5 years, you can't put multi-message select???? wtf!

Makes me not even want to use this crap

**Comment by rubo77** [comment]
It is already possible to react the latest 50 messages or so by a spammer. It is hidden in the right sidebar somewhere

**Reply by iav** [reply]
And it's bad.
Because nobody knows, how many messages will be deleted.

**Comment by kwinz** [comment]
Other issues keep getting closed in favor of this one which are more generally about selecting multiple messages vector-im/element-web#9655, vector-im/element-web#16218.
But here the discussion mainly centers on covenient bulk redacting or deleting! Not surprising because this issue's title is "Ability to redact [...]"

I need multiple message selection for FORWARDING, not for deleting. Please don't forget the FORWARDING use case!

Either also discuss the FORWARDING use case here or allow a separate issue for this use case that is not going to be closed as duplicate.

**Comment by kwinz** [comment]
Can you rename this issue to "Ability to apply actions to multiple messages"? Or "Ability to manipulate multiple messages at once"?
Or should I open a separate issue for that?

As @alittlebitofit said we need an issue that handles

1-multiselect
2-multicopy
3-multidelete
4-multishare
5-other things that can be done with multi feature that i dont recall as of now

Not just redaction.

**Reply by ToddCrimson** [reply]
Plus One (+1) to this. We use matrix in a small server and among conversations, also share media. 

Use case, my Buddy / colleague shared 21 pics.
They each have to be selected and saved.
OR, open to view and then download each one separately. 
Extremely limited support and discourages users from being on this app, and going to Telegram to share media. 

I think @kwinz has the right idea. Let's have an all encompassing ticket.

OR, I sense after reading all these tickets that there is resistance to multi redactions, maybe at the very least allow other multi select options ....

Save media, forward messages etc... 

But you must please enable this very simple integration 
.... Its extremely necessary. This topic has been around now since 2017.

**Comment by damascene** [comment]
Any update on this. what is the convince way to easily delete multiple messages?

**Reply by eandersons** [reply]
Use a Matrix client that supports deleting multiple messages at once. I am using FluffyChat for this purpose.

**Reply by ToddCrimson** [reply]
FluffyChat has its other limitations. 

Why are admins / developers here so resistant to adding that feature? I'm willing to raise money for it. This is a must have, especially how @kwinz stated it.

**Reply by eandersons** [reply]
Yes, FluffyChat has its shortcomings, but the ability to delete multiple messages at once is the only reason and purpose I am still using it, although it has difficulties when deleting more than ~10 messages, but that probably may differ depending on Matrix instance.

Element (its fork actually) is my main Matrix client, and I among others would love the ability to interact with multiple messages at once.

**Comment by altf4arnold** [comment]
Any update on this? We get alarms from our monitoring system every 5 minutes until alarms get acknowledged. That made for some 90 messages some times that I have to then take out for readability of the chat

**Reply by ToddCrimson** [reply]
Plus one !!!!! Can we please get some action or a response / timeline?

**Comment by soda-pop-ice-cream** [comment]
Milestone: 2035
Feel free to screenshot this

**Comment by recolic** [comment]
In every IM, I always keep a chat with myself, to save random texts / notes for later.

But when I need to clear old junks, Element (and all other Matrix client) are so annoying. 

Telegram allow you to multi-select & one-click delete, Teams allow you to right-click + click delete (2 click to delete 1 msg), but Element is the most ridiculous guy: It asks for a reason for each message.

||How many clicks required to delete 30 msg|
|---|---|
|Telegram|3|
|Teams|60|
|Element|90|

**Comment by notDavid** [comment]
Note that moving these feature request from "Issues" to "Discussions" is annoying, because now we can not subscribe to get notified (**only**) when the issue is closed/resolved.

**Comment by Glotzon** [comment]
I just want to ensure this open idea still gets the necessary level of attention in 2026, just 9 years since it was raised.

**Comment by mikebakke** [comment]
Cannot believe (actually I can) that this is not addressed yet - multiple message selection then apply the desired action to selected items. 

Most Matrix Live sessions contains a leader aspiring to wider adoption of the platform but whenever I get a non technical user to try it, issues like this stop them in their tracks because they are used to Telegram, Whatsapp or whatever allowing multi-select - it breaks muscle memory hard.
